bigtable/client/row_range.h

```diff
@@ -0,0 +1,166 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_ROW_RANGE_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_ROW_RANGE_H_
+
+#include "bigtable/client/version.h"
+
+#include <google/bigtable/v2/data.pb.h>
+
+#include <absl/strings/string_view.h>
+
+#include <chrono>
+
+#include "bigtable/client/internal/prefix_range_end.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Define the interfaces to create row key ranges.
+ *
+ * Example:
+ * @code
+ * // Create a range for the keys starting with the given prefix.
+ * auto range = bigtable::RowRange("foo/");
+ * @endcode
+ */
+class RowRange {
+ public:
+  RowRange(RowRange&& rhs) noexcept = default;
+  RowRange& operator=(RowRange&& rhs) noexcept = default;
+  RowRange(RowRange const& rhs) = default;
+  RowRange& operator=(RowRange const& rhs) = default;
+
+  /// Return the infinite range, i.e., a range including all possible keys.
+  static RowRange InfiniteRange() {
+    RowRange tmp;
```

Comment:
I consider `tmp` to be temporary (as the name implies) throwaway objects; what about using `result` (I think you use it elsewhere) for things that are returned?

Reply:
Fixed.